test/DynamoCoreTests/Performance/PerformanceTests.cs

```diff
@@ -112,19 +196,119 @@ private void DoWorkspaceOpenAndCompare(string filePath)
 
+            newEngineMemoryData = StopCollectingMemory(checkPoint);
+
             var wcd2 = new serializationTestUtils.WorkspaceComparisonData(ws2, CurrentDynamoModel.EngineController, dsExecution: false);
 
             serializationTestUtils.CompareWorkspaceModelsMSIL(wcd1, wcd2);
 
             var newEngineCompileAndExecutionTime = model.EngineController.CompileAndExecutionTime;
 
-            Console.WriteLine("Compile and Execution time old Engine={0:0.0}+{1:0.0} ms, new Engine={2:0.0}+{3:0.0} ms",
+            Console.WriteLine("Compile and Execution time old Engine={0:0.0}+{1:0.0} ms, new Engine={2:0.0}+{3:0.0} ms, hardCoded={4:0.0}",
                 oldEngineCompileAndExecutionTime.compileTime.TotalMilliseconds, oldEngineCompileAndExecutionTime.executionTime.TotalMilliseconds,
-                newEngineCompileAndExecutionTime.compileTime.TotalMilliseconds, newEngineCompileAndExecutionTime.executionTime.TotalMilliseconds);
+                newEngineCompileAndExecutionTime.compileTime.TotalMilliseconds, newEngineCompileAndExecutionTime.executionTime.TotalMilliseconds,
+                hardCodedExecutionTime.TotalMilliseconds);
             var execution = (Path.GetFileName(filePath),
                 oldEngineCompileAndExecutionTime.compileTime, oldEngineCompileAndExecutionTime.executionTime,
-                newEngineCompileAndExecutionTime.compileTime, newEngineCompileAndExecutionTime.executionTime);
+                newEngineCompileAndExecutionTime.compileTime, newEngineCompileAndExecutionTime.executionTime,
+                hardCodedExecutionTime);
             executionData.Add(execution);
+
+            CollectMemoryData(Path.GetFileName(filePath));
+        }
+
+        private void CollectMemoryData(string graph)
+        {
+            if (oldEngineMemoryData.collected.SizeInBytes != -1)
+            {
+                var memoryInfo = (graph, oldEngine: oldEngineMemoryData, newEngine: newEngineMemoryData, hardcoded: hardcodedMemoryData);
+                memoryData.Add(memoryInfo);
+            }
+        }
+
+        private void WarmUp()
+        {
+            if (!warmupDone)
+            {
+                var warmupFile = Path.Combine(TestDirectory, "core", "performance", "simple_point.dyn");
+                OpenModel(warmupFile);
+
+                Cleanup();
+
+                Setup();
+
+                OpenModel(warmupFile, dsExecution: false);
+
+                Cleanup();
+
+                Setup();
+
+                warmupDone = true;
+            }
+        }
+
+        private MemoryCheckPoint StartCollectingMemory()
+        {
+            return dotMemory.Check();
+        }
+
+        private (MemoryInfo, MemoryInfo) StopCollectingMemory(MemoryCheckPoint checkPoint)
+        {
+            var retVal = (new MemoryInfo(-1, -1), new MemoryInfo(-1, -1));
+
+            dotMemory.Check(memory =>
+            {
+                var traffic = memory.GetTrafficFrom(checkPoint);
```

Comment:
should we be forcing GC before collecting?

Reply:
I will give it a try and see what happens with the numbers. We can then discuss what makes most sense.